[CI] EsqlClientYamlIT test {p0=esql/50_index_patterns/_index LIKE with overly complex pattern} failing

**Build Scans:**
- [elasticsearch-intake #38992 / 8.19.16_bwc-snapshots](https://gradle-enterprise.elastic.co/s/iai2aazm7nv5e)
- [elasticsearch-pull-request #142604 / 8.19.16_part_3_bwc-snapshots](https://gradle-enterprise.elastic.co/s/w73qbn2cu7vs2)
- [elasticsearch-pull-request #142572 / 8.19.16_part_3_bwc-snapshots](https://gradle-enterprise.elastic.co/s/4y37wnmy4lm54)
- [elasticsearch-pull-request #142572 / 8.19.16_part_3_bwc-snapshots](https://gradle-enterprise.elastic.co/s/zte2qjhldmot4)
- [elasticsearch-intake #38975 / 8.19.16_bwc-snapshots](https://gradle-enterprise.elastic.co/s/ldnf3xd6ldwoa)
- [elasticsearch-intake #38965 / 8.19.16_bwc-snapshots](https://gradle-enterprise.elastic.co/s/q2lfzshcxnzt4)
- [elasticsearch-periodic #12220 / openjdk21_8.19.16_java-fips-matrix-bwc](https://gradle-enterprise.elastic.co/s/5zbkzxrnzyx76)
- [elasticsearch-periodic #12220 / 8.19.16_bwc](https://gradle-enterprise.elastic.co/s/44g7zo4ukjqfu)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:qa:server:mixed-cluster:v8.19.16#yamlRestTest" -Dtests.class="org.elasticsearch.xpack.esql.qa.mixed.EsqlClientYamlIT" -Dtests.method="test {p0=esql/50_index_patterns/_index LIKE with overly complex pattern}" -Dtests.seed=37C91DDFC9C3E835 -Dtests.bwc=true -Dtests.locale=shi-Tfng -Dtests.timezone=Asia/Harbin -Druntime.java=26
```

**Applicable branches:**
9.3

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.qa.mixed.EsqlClientYamlIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('test%20%7Bp0%3Desql%2F50_index_patterns%2F_index%20LIKE%20with%20overly%20complex%20pattern%7D'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: java.lang.AssertionError: the error message was expected to match the provided regex but didn't
Expected: Pattern was too complex to determinize
     but: was "{root_cause=[{type=too_complex_to_determinize_exception, reason=Determinizing automaton with 29 states and 29 transitions would require more than 10000 effort., stack_trace=org.elasticsearch.ElasticsearchException$1: Determinizing automaton with 29 states and 29 transitions would require more than 10000 effort.\n\tat org.elasticsearch.server@8.19.16-SNAPSHOT/org.elasticsearch.ElasticsearchException.guessRootCauses(ElasticsearchException.java:709)\n\tat org.elasticsearch.server@8.19.16-SNAPSHOT/org.elasticsearch.ElasticsearchException.generateFailureXContent(ElasticsearchException.java:637)\n\tat org.elasticsearch.server@8.19.16-SNAPSHOT/org.elasticsearch.rest.RestResponse.build(RestResponse.java:201)\n\tat org.elasticsearch.server@8.19.16-SNAPSHOT/org.elasticsearch.rest.RestResponse.<init>(RestResponse.
[truncated]
```

**Issue Reasons:**
- [9.3] 2 consecutive failures in test test {p0=esql/50_index_patterns/_index LIKE with overly complex pattern}
- [9.3] 1 consecutive failures in step 8.19.16_bwc-snapshots
- [9.3] 3 consecutive failures in step 8.19.16_part_3_bwc-snapshots
- [9.3] 1 consecutive failures in pipeline elasticsearch-intake
- [9.3] 2 consecutive failures in pipeline elasticsearch-pull-request
- [9.3] 8 failures in test test {p0=esql/50_index_patterns/_index LIKE with overly complex pattern} (53.3% fail rate in 15 executions)
- [9.3] 3 failures in step 8.19.16_bwc-snapshots (75.0% fail rate in 4 executions)
- [9.3] 3 failures in step 8.19.16_part_3_bwc-snapshots (60.0% fail rate in 5 executions)
- [9.3] 3 failures in pipeline elasticsearch-intake (75.0% fail rate in 4 executions)
- [9.3] 2 failures in pipeline elasticsearch-pull-request (50.0% fail rate in 4 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -333,9 +333,6 @@ tests:
 - class: org.elasticsearch.xpack.esql.plugin.RerankIT
   method: testRerankRowLimitEnforcement
   issue: https://github.com/elastic/elasticsearch/issues/148102
-- class: org.elasticsearch.xpack.esql.qa.mixed.EsqlClientYamlIT
-  method: test {p0=esql/50_index_patterns/_index LIKE with overly complex pattern}
-  issue: https://github.com/elastic/elasticsearch/issues/148134
 - class: org.elasticsearch.xpack.security.support.MetadataFlattenedMigrationIntegTests
   method: testMigrationWithConcurrentUpdates
   issue: https://github.com/elastic/elasticsearch/issues/143674

--- a/x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/esql/50_index_patterns.yml
+++ b/x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/esql/50_index_patterns.yml
@@ -404,7 +404,7 @@ same_name_different_type_same_family:
         - method: POST
           path: /_query
           parameters: [ method, path, parameters, capabilities ]
-          capabilities: [ fix_index_like_question_mark_wildcard ]
+          capabilities: [ fix_index_like_question_mark_wildcard, handle_determinization_complexity ]
       reason: "determinization complexity for _index LIKE not handled until fix"
   - do:
       indices.create:

